datahub-web-react/src/app/permissions/policy/ManagePolicies.tsx

```diff
@@ -301,23 +316,55 @@ export const ManagePolicies = () => {
     const onSavePolicy = (savePolicy: Omit<Policy, 'urn'>) => {
         if (focusPolicyUrn) {
             // If there's an URN associated with the focused policy, then we are editing an existing policy.
-            updatePolicy({ variables: { urn: focusPolicyUrn, input: toPolicyInput(savePolicy) } });
-            analytics.event({
-                type: EventType.UpdatePolicyEvent,
-                policyUrn: focusPolicyUrn,
+            updatePolicy({ variables: { urn: focusPolicyUrn, input: toPolicyInput(savePolicy) } })
+            .then(() => {
+                analytics.event({
+                    type: EventType.UpdatePolicyEvent,
+                    policyUrn: focusPolicyUrn,
+                });
+                message.success({
+                    content: `Successfully updated policy!`,
+                    duration: 3,
+                });
+                policiesRefetch();
+                onClosePolicyBuilder();
+            })
+            .catch((e) => {
+                message.destroy();
+                message.error({
+                    content: `Failed to update policy!: \n ${e.message || ''}`,
+                    duration: 3,
+                });
             });
+            
         } else {
-            // If there's no URN associated with the focused policy, then we are creating.
-            createPolicy({ variables: { input: toPolicyInput(savePolicy) } });
-            analytics.event({
-                type: EventType.CreatePolicyEvent,
-            });
+            createPolicy({ variables: { input: toPolicyInput(savePolicy) }  })
+                .then((result) => {
+                    const newPolicy = {
+                        urn: result?.data?.createPolicy,
+                        ...savePolicy,
+                    };
+                    addToListPoliciesCache(client, newPolicy, pageSize, query);
+                    setTimeout(() => {
+                        policiesRefetch();
+                        analytics.event({
+                            type: EventType.CreatePolicyEvent,
+                        });
+                        message.success({
+                            content: `Successfully saved policy.`,
+                            duration: 3,
+                        })
+                    }, 2000);
+                    onClosePolicyBuilder();
+                })
+                .catch((e) => {
+                    message.destroy();
+                    message.error({
+                        content: `Failed to create policy!: \n ${e.message || ''}`,
```

Comment:
Let's just say: Failed to create policy! An unexpected error occurred.

Reply:
Done